Automattic/woocommerce-payments
# Reduce unit test dependencies

Fixes #86 

#### Changes proposed in this Pull Request

A few people have expressed interest in exploring what we would need to do to make our unit tests quicker to run and easier to setup. This PR starts exploring the options.

Firstly we move everything into a "unit" folder, which gives us space to use the existing bootstrap and setup for full e2e tests.

The commit messages describe some of the challenges we'll hit in trying to do this, along with proposed solutions. I'd be really interested in what people think.

#### Testing instructions

1. Clone the project
2. Run `composer install`
3. Run `./vendor/bin/phpunit'
4. The tests should run and pass without a WordPress DB or environment running

#### Questions and Notes

1. I had to upgrade PHPUnit to use the WP_Mock package. This isn't ideal given the PHP versions we're trying to support.
2. Things are failing on the Travis builds. Ultimately this work should simplify our CI (for unit tests) but things will need tweaking. All the other work on our CI pipeline will still be needed and relevant for running e2e tests. What this does give us is quicker unit tests to flush out easy to find problems sooner.
